modules/jwt_auth_refresh/src/Controller/RefreshController.php

```diff
@@ -0,0 +1,156 @@
+use Drupal\Core\Flood\FloodInterface;
+use Drupal\Core\Session\AccountSwitcherInterface;
+use Drupal\jwt\Authentication\Provider\JwtAuth;
+use Drupal\jwt\Transcoder\JwtDecodeException;
+use Drupal\jwt\Transcoder\JwtTranscoderInterface;
+use Drupal\jwt_auth_issuer\Controller\JwtAuthIssuerController;
+use Drupal\jwt_auth_refresh\JwtRefreshTokensInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+class RefreshController extends JwtAuthIssuerController {
+
+  /**
+   * Account switcher service.
+   *
+   * @var \Drupal\Core\Session\AccountSwitcherInterface
+   */
+  protected $accountSwitcher;
+
+  /**
+   * Flood service.
+   *
+   * @var \Drupal\Core\Flood\FloodInterface
+   */
+  protected $flood;
+
+  /**
+   * JWT Transcoder.
+   *
+   * @var \Drupal\jwt\Transcoder\JwtTranscoderInterface
+   */
+  protected $jwtTranscoder;
+
+  /**
+   * Current request.
+   *
+   * @var null|\Symfony\Component\HttpFoundation\Request
+   */
+  protected $currentRequest;
+
+  /**
+   * @inheritDoc
+   */
+  public function __construct(JwtAuth $auth, JwtRefreshTokensInterface $refreshTokens, AccountSwitcherInterface $accountSwitcher, FloodInterface $flood, JwtTranscoderInterface $jwtTranscoder, RequestStack $requestStack) {
+    parent::__construct($auth, $refreshTokens);
+    $this->accountSwitcher = $accountSwitcher;
+    $this->flood = $flood;
+    $this->jwtTranscoder = $jwtTranscoder;
+    $this->currentRequest = $requestStack->getCurrentRequest();
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('jwt.authentication.jwt'),
+      $container->get('jwt_auth_refresh.tokens'),
+      $container->get('account_switcher'),
+      $container->get('flood'),
+      $container->get('jwt.transcoder'),
+      $container->get('request_stack')
+    );
+  }
+
+  /**
+   * Enforces flood control for the current login request, by IP.
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException
+   */
+  protected function floodControl() {
+    $flood_config = $this->config('user.flood');
+    if (!$this->flood->isAllowed('jwt_auth_refresh.failed_refresh_ip', $flood_config->get('ip_limit'), $flood_config->get('ip_window'))) {
+      throw new AccessDeniedHttpException('Access is blocked because of IP based flood prevention.', NULL, Response::HTTP_TOO_MANY_REQUESTS);
+    }
+  }
+
+  /**
+   * Refresh controller.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   * @return \Symfony\Component\HttpFoundation\JsonResponse
+   */
+  public function refresh(Request $request) {
+    $owner = $this->getToken($request)->getOwner();
+    // @todo - Better approach than switching?
```

Comment:
I see. Yeah, I think that the event subscriber model should go. It was over-engineered and I don't think it provides a lot of benefit. I'd like to see the JwtToken class implement `create()` and `__toString()`. Thus, generating a token would look something like:

```
$jwt_string = (string) JwtToken::create(['payload' => ['key' => 'value']]);
```